### PR TITLE
was/attachments

### PR DESCRIPTION
**Pull request by ngaglani-tenable**

### CODE REVIEW

**Review by parvdesai-tenable — APPROVED** [review]

File: tenable/io/v3/was/attachments/api.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+class AttachmentsAPI(ExploreBaseEndpoint):
+
+    _path = 'api/v3/was/attachments'
+
+    def download_attachment(
+        self,
+        attachment_id: UUID,
+        fobj: Optional[BinaryIO] = None
+    ) -> BinaryIO:
```
Comment (sahils-tenable) [inline tenable/io/v3/was/attachments/api.py:29]: Hi nishant, could u pls update the parameters as per the latest standard => please pick the first approach = https://github.com/dhvanilshah-tenable/pyTenable/pull/12#discussion_r769828173

File: tests/io/v3/was/test_attachments_api.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+@responses.activate
+def test_download_img_attachment(api):
+
+    global IMG_CONTENTS
+
+    if IMG_CONTENTS is None:
+        with open(IMG_ATTACHMENT_FILE_PATH, 'rb') as img_file:
+            IMG_CONTENTS = img_file.read()
+
```
Comment (sahils-tenable) [inline tests/io/v3/was/test_attachments_api.py:35]: can we try to write this logic in a separate method, as we are using it at multiple place

**Review by sahils-tenable — APPROVED** [review]

File: tenable/io/v3/was/attachments/api.py (hunk 1 above)
Comment (ngaglani-tenable) [reply tenable/io/v3/was/attachments/api.py]: Done

File: tests/io/v3/was/test_attachments_api.py (hunk 2 above)
Comment (ngaglani-tenable) [reply tests/io/v3/was/test_attachments_api.py]: Done.

File: tenable/io/v3/was/attachments/api.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+'''
+from io import BytesIO
+from typing import BinaryIO, Optional
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+
+
+class AttachmentsAPI(ExploreBaseEndpoint):
+
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/was/attachments/api.py:22]: '''
    This class contains methods related to Attachments API
    '''

File: tenable/io/v3/was/attachments/api.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        '''
+        Download an attachment.
+
+        :devportal:`attachments: download <was-v2-attachments-download>`
+
+        Args:
+            attachment_id:
+                The unique identifier for attachment.
+            fobj:
+                A file-like object to write the contents of the policy to.  If
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/was/attachments/api.py:38]: A file-like object to write the contents of the attachment to

File: tests/io/v3/was/attachments/test_attachments_api.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    if TXT_CONTENTS is None:
+        with open(TXT_ATTACHMENT_FILE_PATH, 'rb') as txt_file:
+            TXT_CONTENTS = txt_file.read()
+
+
+@responses.activate
+def test_download_img_attachment(api):
+
+    generate_image_contents()
```
Comment (dhvanilshah-tenable) [inline tests/io/v3/was/attachments/test_attachments_api.py:48]: You could call generate_image_contents and generate_text_contents more cleanly as a fixture. Ref: https://github.com/dhvanilshah-tenable/pyTenable/pull/39/files#diff-fa3961de3a9de1ad4460055d8ba0cedd004c7c1355d983724362fdeba568cfa7R8

**Review by dhvanilshah-tenable — APPROVED** [review]

File: tests/io/v3/was/attachments/test_attachments_api.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        body=TXT_CONTENTS
+    )
+
+    resp = api.v3.was.attachments.download_attachment(TXT_ATTACHMENT_FILE)
+    assert resp.read() == TXT_CONTENTS
+
+
+@responses.activate
+def test_download_txt_file_attachment(api):
+
```
Comment (dhvanilshah-tenable) [inline tests/io/v3/was/attachments/test_attachments_api.py:101]: Please add doc strings for the tests.

File: tenable/io/v3/was/attachments/api.py (hunk 3 above)
Comment (ngaglani-tenable) [reply tenable/io/v3/was/attachments/api.py:22]: Done.

File: tenable/io/v3/was/attachments/api.py (hunk 4 above)
Comment (ngaglani-tenable) [reply tenable/io/v3/was/attachments/api.py]: Done.

File: tests/io/v3/was/attachments/test_attachments_api.py (hunk 6 above)
Comment (ngaglani-tenable) [reply tests/io/v3/was/attachments/test_attachments_api.py:113]: Done.

Comment (ngaglani-tenable) [reply tests/io/v3/was/attachments/test_attachments_api.py:113]: Done.

File: tests/io/v3/was/attachments/test_attachments_api.py (hunk 5 above)
Comment (ngaglani-tenable) [reply tests/io/v3/was/attachments/test_attachments_api.py]: Done.

**Review by tomeara — APPROVED** [review]

**Pull request merged**